### DISCUSSION TITLE
Create multiple elements within loop?

### DISCUSSION THREAD

**Discussion by HansenBerlin** [framesurge/perseus#255] [Q&A]
Hi, 

I am new to Rust and therefore this library (.NET, JS, Java, Python till now). I couldnt figure out (and documentation is pretty sparse) how to add an element multiple times. This for example:

![image](https://user-images.githubusercontent.com/42807459/215362463-a74165e2-0932-4cfb-89a7-1481d099017e.png)

How can I add the card div say 10 times within the features wrapper either using a loop within the markup code or value of a function returning this piece of markup (if possible).?

Thanks and keep up the great work (pretty straightforward experience so far except of the docs. ;-) )

**Comment by HansenBerlin** [comment]
Got it :-)
![image](https://user-images.githubusercontent.com/42807459/215366813-c80822dd-358c-4032-a931-a7e4a1109f8b.png)

call:
![image](https://user-images.githubusercontent.com/42807459/215366827-b95fedc1-45df-4ce0-b84a-7c7efae8bd02.png)

(Just in case anyone struggles with the same problem)

**Reply by lukechu10** [reply]
Hey there, I saw you created a new discussion on Sycamore as well. The solution you came up with certainly works but I would suggest using the `Indexed` component instead, which was built specifically for this. Since you are just rendering the same identical element multiple times, you can just do
```rust
Indexed(
    iterable=create_signal(cx, (0..n).collect::<Vec<_>>()),
    view=|cx, _| view! { cx,
        div(...) {...}
    }
}
```
The reason for this is because this is a more declarative way of writing your view code. You describe directly how you want your final view to look like instead of manually constructing it using `for` loops and temporary variables.

Hope this helps!
